apps/tools/components/send_invitation/index.coffee

```diff
@@ -8,12 +8,42 @@ module.exports = ($el) ->
   $submit = $el.find '.js-submit'
   $errors = $el.find '.js-errors'
 
+  submissionTimeout = null
+  label = $submit.text()
+
+  onSuccess = () ->
+    $form.trigger 'reset'
+
+    $submit
+      .prop 'disabled', false
+      .text 'Sent!'
+
+    submissionTimeout = setTimeout (-> $submit.text label), 2500
+
+    track.submit 'Invitation sent from user'
+
+  onFailure = (message) ->
+    $errors
+      .show()
+      .html message
+
+    $submit
+      .prop 'disabled', false
+      .text 'Error'
+
+    submissionTimeout = setTimeout ->
+      $submit.text label
+      $errors.empty()
+    , 5000
+
+    track.error 'Invitation not sent, try again.'
+
   $form.on 'submit', (e) ->
     e.preventDefault()
 
-    serializer = new Serializer $form
+    submissionTimeout && clearTimeout(submissionTimeout);
```

Comment:
Since this is a .coffee file we might as well be idiomatic; (I realize there's no linter here) (`&& => and`, no semi-colons).

Also `clearTimeout` will return fine on null/undefined, so there's no need to do the existence check.